api/loader.go

```diff
@@ -63,10 +66,17 @@ func (l *loader) run(ctx context.Context) error {
 
 			log.Debugf("got job details: %+v", job)
 			l.jobsCache.Cache[j] = job
+			cache[j] = job
+		}
+	}
+
+	for k := range l.jobsCache.Cache {
+		if _, ok := cache[k]; !ok {
+			delete(l.jobsCache.Cache, k)
```

Comment:
looping through above adds all of the _new_ jobs to the cache, but doesn't remove jobs that have been deleted.  this `cache` does garbage collection on the deleted jobs.  I played with replacing the cache atomically, but it wasn't straight forward since the map is shared across a bunch of running routines.  this could probably be more elegant...

Reply:
oh, i see it now 👍